adagrams/game.py

```diff
@@ -1,11 +1,133 @@
+    # iterates through each capitalized letter in the word the user passes in
+    for letter in word.upper():
+        # checks if the current letter if found in the letter_bank_copy list
+        if letter in letter_bank_copy:
+            # if the current letter is found, remove that instance from the list
+            letter_bank_copy.remove(letter)
+        else:
+            # on the first instance the current letter is not found in the letter_bank_copy list, return False
+            return False
+    # ends function and returns True if all of the letters in word are found in the letter_bank list
+    return True
+
 
 def score_word(word):
-    pass
+    # SCORE_CHART is a chart that contains the point value for each letter
+    SCORE_CHART = {
+        "AEIOULNRST": 1,
+        "DG": 2,
+        "BCMP": 3,
+        "FHVWY": 4,
+        "K": 5,
+        "JX": 8,
+        "QZ": 10
+    }
+    # initializes score variable that will keep track of the accumulating score for each letter found in the user's word
+    score = 0
+    # grabs the key and value of each dicionary key-pair in the SCORE_CHART dictionary
+    for key, value in SCORE_CHART.items():
+        # iterate through each letter in capitalized word
+        for letter in word.upper():
+            # searches for each letter in a key of letters from SCORE_CHART
+            if letter in key:
+                # if the current letter is found in any of the keys from SCORE_CHART, add that key's value to the score
+                score += value
+    # checks if the length of the word is between 7 and 10
+    if 7 <= len(word) <= 10:
+      # if the length of the word is between 7 and 10, an additional 8 bonus points are added to score
+        score += 8
+    # returns the final score
+    return score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    # initialize best_word variable that will hold the best_word
+    best_word = ""
+    # initialize highest_score variable that will hold the highest_score
+    highest_score = 0
+    # iterate through numvers 0 to the length of the word_list
+    for i in range(0, len(word_list)):
+        # grab the score for the current word iteration by calling score_word function
+        current_word_score = score_word(word_list[i])
+        # checks if the current_word_score iteration is greater than the highest_score
+        if current_word_score > highest_score:
+            # if current_word_score is greater than highest_score, update best_word and highest_score
+            best_word = word_list[i]
+            highest_score = current_word_score
+        # else if current_word_scre equals highest_score:
+        elif current_word_score == highest_score:
+            # if current_word_score is equal to highest_score, keep first instance
+            if (len(word_list[i]) == 10) and (len(best_word) == 10):
+                best_word = best_word
+                highest_score = highest_score
```

Comment:
Hm, do we need this conditional? We are re-assigning the current highest scored word with itself. Let's get rid of it.